float64 x
float64 y
float64 z
---
joint_angles joint_vals